src/rgw/driver/rados/rgw_neorados.h

```diff
@@ -0,0 +1,54 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#pragma once
+
+#include <coroutine>
+#include <string_view>
+
+#include <boost/asio/awaitable.hpp>
+#include <boost/asio/async_result.hpp>
+#include <boost/asio/use_awaitable.hpp>
+
+#include <boost/system/detail/errc.hpp>
+
+#include "include/neorados/RADOS.hpp"
+
+#include "common/dout.h"
+
+#include "osd/osd_types.h"
+
+#include "rgw_obj_types.h"
+#include "rgw_pool_types.h"
+
+namespace rgw::neorados {
+using namespace ::neorados;
```

Comment:
I'm going to switch these to co_compose versions, so we don't lock ourselves to the choice of executor, since I ran into that on the metadata refactor.

But also move them to `rgw_tools.h` under the `rgw` namespace.